### PR TITLE
Mockserver

### PR DESCRIPTION
**Pull request by Garlyle**

### CODE REVIEW

File: app/src/main/java/com/greenfox/opal/gitinder/response/StatusResponse.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.greenfox.opal.gitinder.response;
+
+public class StatusResponse {
+    String status;
+    String token;
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/response/StatusResponse.java]: @Garlyle status response doesnt contain a token a LoginResponse should

File: app/src/main/java/com/greenfox/opal/gitinder/response/LoginResponse.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.greenfox.opal.gitinder.response;
+
+public class LoginResponse extends StatusResponse {
+    public LoginResponse(String status, String token) {
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/response/LoginResponse.java]: @Garlyle there is no need for these constructors (unless you are using it somewhere directly), just an empty one (json conversion will need that)

File: app/src/main/java/com/greenfox/opal/gitinder/service/MockCall.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.greenfox.opal.gitinder.service;
+
+import java.io.IOException;
+
+import okhttp3.Request;
+import retrofit2.Call;
+import retrofit2.Response;
+
+public abstract class MockCall implements Call {
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/service/MockCall.java]: @Garlyle make the MockCall generic using `<T>`

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/greenfox/opal/gitinder/response/StatusResponse.java (hunk 1 above)
Comment (Garlyle) [inline app/src/main/java/com/greenfox/opal/gitinder/response/StatusResponse.java]: StatusResponse is the class we mentioned that stores all possible variables. Other responses will inherit from that.

File: app/src/main/java/com/greenfox/opal/gitinder/response/LoginResponse.java (hunk 2 above)
Comment (Garlyle) [inline app/src/main/java/com/greenfox/opal/gitinder/response/LoginResponse.java]: Currently the MockServer is using them to create responses.

File: app/src/main/java/com/greenfox/opal/gitinder/ApiService.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import com.greenfox.opal.gitinder.model.LoginRequest;
+import com.greenfox.opal.gitinder.service.MockCall;
+
+import retrofit2.Call;
+import retrofit2.http.Body;
+import retrofit2.http.POST;
+
+public interface ApiService {
+    @POST("/login")
+    Call login(@Body LoginRequest loginRequest);
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/ApiService.java]: @Garlyle this should return `Call<LoginResponse>`

File: app/src/main/java/com/greenfox/opal/gitinder/service/MockServer.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.http.Body;
+
+public class MockServer implements ApiService {
+    @Override
+    public Call login(@Body final LoginRequest loginRequest) {
+        return new MockCall() {
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/opal/gitinder/service/MockServer.java]: @Garlyle this should be also `MockCall<LoginResponse>`

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**